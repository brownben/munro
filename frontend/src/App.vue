<template>
  <a
    href="#content"
    class="
      z-50
      bg-white
      sr-only
      focus:absolute
      focus:px-3
      focus:py-2
      focus:m-4
      text-main-800
      rounded-shape
      focus:not-sr-only
      focus:block
      focus:shadow-outline
    "
  >
    Skip to main content
  </a>
  <AppMenu v-if="!$route.path.includes('/embed/')" />

  <div class="flex flex-col flex-grow">
    <Messages />
    <router-view v-slot="{ Component }">
      <transition name="fade" mode="out-in">
        <component :is="Component" />
      </transition>
    </router-view>
  </div>
</template>

<script setup lang="ts">
import { watchEffect } from 'vue'
import { useRoute } from 'vue-router'

import { toSingleString } from './scripts/typeHelpers'

import AppMenu from './components/AppMenu.vue'
import Messages from './components/AppMessages.vue'

const route = useRoute()

watchEffect(async () => {
  if (route.query.theme) {
    const setTheme = (await import('./setThemes')).default
    setTheme(toSingleString(route.query.theme))
  }
})
</script>
