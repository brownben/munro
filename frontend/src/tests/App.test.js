/*
  App.vue Unit Tests
*/

import { shallowMount } from '@vue/test-utils'
import App from '@/App'

test('Is a Vue Instance', () => {
  const wrapper = shallowMount(App, {
    mocks: {
      $auth: { checkLogin: jest.fn(), user: false },
    },
    stubs: ['router-view'],
  })
  expect(wrapper.isVueInstance()).toBeTruthy()
})

test('Renders Correctly', () => {
  const wrapper = shallowMount(App, {
    mocks: {
      $auth: { checkLogin: jest.fn(), user: false },
    },
    stubs: ['router-view'],
  })
  expect(wrapper.element).toMatchSnapshot()
})
