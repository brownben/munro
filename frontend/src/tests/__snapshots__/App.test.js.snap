// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<div
  class="flex flex-col w-full h-full"
  id="app"
>
  <appmenu-stub />
   
  <div
    class="flex-grow"
    id="content"
  >
    <messages-stub />
     
    <router-view-stub
      class=""
      id="router-view"
      mode="out-in"
      name="fade"
    />
  </div>
   
  <appfooter-stub />
</div>
`;
